File: client-scalaz/src/main/scala/org/openapitools/client/api/QuestionResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import QuestionResponse._

case class QuestionResponse (
  id: Option[Long],
question: Option[String],
image: Option[AssetShortResponse],
videoURL: Option[String],
active: Option[Boolean],
answers: Option[List[AnswerResponse]],
scores: Option[ScoreListResponse],
tickets: Option[TicketListResponse],
owner: Option[AccountShortResponse],
icon: Option[AssetShortResponse],
authorOverride: Option[String],
updatedDate: Option[Long],
startDate: Option[Long],
endDate: Option[Long],
createdDate: Option[Long],
tags: Option[String],
allocateTickets: Option[Boolean],
ticketType: Option[String],
ticketCount: Option[Long],
points: Option[Long])

object QuestionResponse {
  import DateTimeCodecs._

  implicit val QuestionResponseCodecJson: CodecJson[QuestionResponse] = CodecJson.derive[QuestionResponse]
  implicit val QuestionResponseDecoder: EntityDecoder[QuestionResponse] = jsonOf[QuestionResponse]
  implicit val QuestionResponseEncoder: EntityEncoder[QuestionResponse] = jsonEncoderOf[QuestionResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/question_response.clj
(ns sirqul-io-t-platform.specs.question-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.answer-response :refer :all]
            [sirqul-io-t-platform.specs.score-list-response :refer :all]
            [sirqul-io-t-platform.specs.ticket-list-response :refer :all]
            [sirqul-io-t-platform.specs.account-short-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            )
  (:import (java.io File)))


(def question-response-data
  {
   (ds/opt :id) int?
   (ds/opt :question) string?
   (ds/opt :image) asset-short-response-spec
   (ds/opt :videoURL) string?
   (ds/opt :active) boolean?
   (ds/opt :answers) (s/coll-of answer-response-spec)
   (ds/opt :scores) score-list-response-spec
   (ds/opt :tickets) ticket-list-response-spec
   (ds/opt :owner) account-short-response-spec
   (ds/opt :icon) asset-short-response-spec
   (ds/opt :authorOverride) string?
   (ds/opt :updatedDate) int?
   (ds/opt :startDate) int?
   (ds/opt :endDate) int?
   (ds/opt :createdDate) int?
   (ds/opt :tags) string?
   (ds/opt :allocateTickets) boolean?
   (ds/opt :ticketType) string?
   (ds/opt :ticketCount) int?
   (ds/opt :points) int?
   })

(def question-response-spec
  (ds/spec
    {:name ::question-response
     :spec question-response-data}))
